File: typescript/src/data_models/manual_task/manual_task.ts
/**
 * Describes a MaualTask that belongs to an active correlation.
 */
export class ManualTask {
  /**
   * The Model ID of the ManualTask as it is declared in the ProcessModel.
   */
  public id: string;
  /**
   * The name of the ManualTask as it is declared in the ProcessModel.
   */
  public name: string;
  /**
   * The Instance ID of the ManualTask.
   */
  public flowNodeInstanceId?: string;
  /**
   * The ID of the correlation that the ManualTask belongs to.
   */
  public correlationId: string;
  /**
   * The ID of the ProcessModel that the ManualTask belongs to.
   */
  public processModelId: string;
  /**
   * The ID of the ProcessInstance that the ManualTask belongs to.
   */
  public processInstanceId?: string;
  /**
   * The token payload the ManualTask got suspended with.
   */
  public tokenPayload: any;
}

File: dotnet/src/data_models/manual_task/ManualTask.cs
namespace ProcessEngine.ConsumerAPI.Contracts.DataModel
{
    /// <summary>
    /// Describes a MaualTask that belongs to an active correlation.
    /// </summary>
    public class ManualTask
    {
        /// <summary>
        /// The ID of the ManualTask.
        /// </summary>
        public string Id { get; set; }

        /// <summary>
        /// The FlowNodeInstance ID of the ManualTask.
        /// </summary>
        public string FlowNodeInstanceId { get; set; }

        /// <summary>
        /// The name of the ManualTask.
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// The ID of the Correlation that the ManualTask belongs to.
        /// </summary>
        public string CorrelationId { get; set; }

        /// <summary>
        /// The ID of the ProcessModel that the ManualTask belongs to.
        /// </summary>
        public string ProcessModelId { get; set; }

        /// <summary>
        /// The ID of the ProcessInstance that the ManualTask belongs to.
        /// </summary>
        public string ProcessInstanceId { get; set; }

        /// <summary>
        /// The token payload the ManualTask got suspended with.
        /// </summary>
        public object TokenPayload { get; set; }
    }
}
